File: java/src/main/java/com/pahappa/systems/commssdk/v1/models/WalletType.java
package com.pahappa.systems.commssdk.v1.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.Getter;

@Getter
public enum WalletType {
    LOCAL("Local"),
    INTERNATIONAL("International");

    private final String value;

    WalletType(String value) {
        this.value = value;
    }

    @Override @JsonValue
    public String toString() {
        return value;
    }

    @JsonCreator
    public static WalletType fromValue(String text) {
        for (WalletType type : WalletType.values()) {
            if (type.value.equals(text)) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown wallet type value: " + text);
    }
}

File: kotlin/src/main/kotlin/v1/models/WalletType.kt
package v1.models

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class WalletType(private val value: String) {
    LOCAL("Local"),
    INTERNATIONAL("International");

    @JsonValue
    override fun toString(): String {
        return value
    }

    companion object {
        @JsonCreator
        fun fromValue(text: String?): WalletType {
            for (type in WalletType.entries) {
                if (type.value == text) {
                    return type
                }
            }
            throw IllegalArgumentException("Unknown wallet type value: $text")
        }
    }
}
